docs: rewrite Actor-to-Actor integration page (#2546)
apify/apify-docs: sources/platform/integrations/actors/index.md

---

> title: What are Actor integrations?

Reviewer: Why ask when we are explaining.

```suggestion
title: What are Actor integrations
```

:::

> Actor-to-Actor integrations connect your Actors with other Actors or tasks, letting you build multi-step workflows on the Apify platform. Because any [integration-ready Actor](/platform/integrations/actors/integration-ready-actors) can act as the target of an integration, the community can keep extending the catalog with new integrations.

Reviewer: I know that tasks are just pre-filled Actors in the end but that sentence bother me :D I doubt tho there is anything good to do about it ¯\_(ツ)_/¯

:::

Actor integrations connect your Actors with other Actors or tasks, letting you build multi-step workflows on the Apify platform. Because any [integration-ready Actor](/platform/integrations/actors/integration-ready-actors) can act as the target of an integration, the community can keep extending the catalog with new integrations.

> <!-- TODO: confirm the YouTube video (zExnYbvFoBM) still reflects the current flow; replace or remove if outdated. -->

Reviewer: that is a terrible spell check :D, also I highly doubt I highly doubt that the video is still relevant. I would nuke it, if by some chance it is okay I still would demote it from full on embed to just link. And definitely place it differently so it isn't before the prose that has highest chance ATM to not e stale

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/zExnYbvFoBM" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

## Add an integration

To connect an Actor with another Actor or task:

1. Open the Actor's detail page and select the **Integrations** tab.

>     - If the Actor already has integrations set up, the **Integrations** tab lists them instead. Click **Add integration** to open the catalog.

Reviewer: Wouldn't it be better if instead of nestd bullet that would be a regular prose below the list?

Author: This way it's directly in the place where you'd need the info

1. In the **Add integration** catalog, find the target Actor in one of two places:
    - **Suggested for this Actor** - Actors relevant to the source Actor's output. <!-- TODO: confirm how an Actor ends up in this list (Store category, manual curation, similarity, or a combination). -->
    - **Search integrations** field at the top - search across the full catalog.
1. Click the Actor card to open its setup screen. <!-- TODO: confirm the post-selection screen still uses the trigger + payload + variables layout described below. Recapture screenshots if the UI differs. -->

> <!-- TODO: add a screenshot of the new Add integration dialog. -->

Reviewer: If we ship with screenshots we'll be doing double duty later when remaking them into figma artifacts, but this page has already surfaced few times so lets ship it with screenshot and change to svgs later

On the setup screen, configure:

- **Triggers** - Events that trigger the integrated Actor. These match webhook [event types](/platform/integrations/webhooks/events) (`run succeeded`, `build failed`, and so on).

    ![Integration trigger select](./images/integration_triggers.png)

- **Input for the integrated Actor** - The input usually has two parts: static fields whose value doesn't change between runs (for example, a database connection string or table name) and dynamic fields specific to the triggering run (for example, a dataset ID). Static fields go in as-is. Dynamic fields are pulled from the implicit `payload` field (the default for most integration-ready Actors) or set with variables.

> - **Available variables** - The same variables as in webhooks. The most common is `{{resource}}`, which holds the Run object in the shape returned by the [API](/api/v2/actor-run-get), or the Build object for build event types. Variables support dot notation, so `{{resource.defaultDatasetId}}` or `{{resource.defaultKeyValueStoreId}}` is usually all you need.

Reviewer: Usually just feels wrong, it implies some unusual cases that should be enumerated, but are not.

```suggestion
- **Available variables** - The same variables as in webhooks. The most common is `{{resource}}`, which holds the Run object in the shape returned by the [API](/api/v2/actor-run-get), or the Build object for build event types. Variables support dot notation, `{{resource.defaultDatasetId}}` and `{{resource.defaultKeyValueStoreId}}` cover most cases.
```
